pkg/controller/erspan.go

```diff
@@ -0,0 +1,358 @@
+		if exists && err == nil {
+			cont.queuePodUpdate(podobj.(*v1.Pod))
+		}
+	})
+	spanupdate := func(spankey string) {
+		spanobj, exists, err := cont.erspanIndexer.GetByKey(spankey)
+		if exists && err == nil {
+			cont.queueErspanUpdate(spanobj.(*erspanpolicy.ErspanPolicy))
+		}
+	}
+	spanhash := func(pod *v1.Pod) string {
+		return pod.Status.PodIP
+	}
+	// Callback when pods selected by erspan object change
+	cont.erspanPolPods.SetObjUpdateCallback(spanupdate)
+	cont.erspanPolPods.SetPodHashFunc(spanhash)
+}
+
+func (cont *AciController) handleErspanUpdate(span *erspanpolicy.ErspanPolicy) bool {
+
+	erspanLogger := ErspanPolicyLogger(cont.log, span)
+	spankey, err := cache.MetaNamespaceKeyFunc(span)
+	if err != nil {
+		erspanLogger.Error("Could not create erspan policy key: ", err)
+		return false
+	}
+	labelKey := cont.aciNameForKey("span", spankey)
+	cont.log.Debug("creating erspanpolicy")
+
+	// Source policies
+	srcGrp := apicapi.NewSpanVSrcGrp(labelKey)
+	srcName := labelKey + "_Src"
+	apicSlice := apicapi.ApicSlice{srcGrp}
+	if span.Spec.Source.AdminState != "" {
+		srcGrp.SetAttr("adminSt", span.Spec.Source.AdminState)
+	} else {
+		srcGrp.SetAttr("adminSt", "start")
+	}
+	src := apicapi.NewSpanVSrc(srcGrp.GetDn(), srcName)
+	srcGrp.AddChild(src)
+	if span.Spec.Source.Direction != "" {
+		src.SetAttr("dir", span.Spec.Source.Direction)
+	} else {
+		src.SetAttr("dir", "both")
+	}
+
+	// Build fvCEp for matching pods
+	cont.indexMutex.Lock()
+	podKeys := cont.erspanPolPods.GetPodForObj(spankey)
+	cont.log.Debug("podkeys are: ", podKeys)
+	for _, podkey := range podKeys {
+		cont.log.Debug("podkey is: ", podkey)
+		if podkey, ok := cont.podIftoEp[podkey]; !ok {
+			cont.log.Warning("Could not find podif data for ", podkey)
+			continue
+		}
+		macRaw := cont.podIftoEp[podkey].MacAddr
+		mac := strings.ToUpper(macRaw)
+		epg := cont.podIftoEp[podkey].EPG
+		appProfile := cont.podIftoEp[podkey].AppProfile
+		fvCEpDn := fmt.Sprintf("uni/tn-%s/ap-%s/epg-%s/cep-%s",
+			cont.config.AciPolicyTenant, appProfile, epg, mac)
+		srcCEp := apicapi.NewSpanRsSrcToVPort(src.GetDn(), fvCEpDn)
+		src.AddChild(srcCEp)
+	}
+	cont.indexMutex.Unlock()
+
+	// Destination policies
+	destGrp := apicapi.NewSpanVDestGrp(labelKey)
+	destName := labelKey + "_Dest"
+	dest := apicapi.NewSpanVDest(destGrp.GetDn(), destName)
+	destGrp.AddChild(dest)
+	destSummary := apicapi.NewSpanVEpgSummary(dest.GetDn())
+	dest.AddChild(destSummary)
+	destSummary.SetAttr("dstIp", span.Spec.Dest.DestIP)
+	if span.Spec.Dest.FlowID != 0 {
+		destSummary.SetAttr("flowId", strconv.Itoa(span.Spec.Dest.FlowID))
+	} else {
+		destSummary.SetAttr("flowId", "1")
+	}
+	apicSlice = append(apicSlice, destGrp)
+
+	var paths []string
+	cont.indexMutex.Lock()
+	for node := range cont.nodeOpflexDevice {
+		fabricPath, ok := cont.fabricPathForNode(node)
+		if !ok {
+			continue
+		}
+		paths = append(paths, fabricPath)
+	}
+	cont.indexMutex.Unlock()
+
+	var vpcs []string
```

Comment:
What if the opflexODev is on an access port?

Reply:
Added support for Leaf Access Ports in the latest commit